frontend/public/components/modals/_modals.scss

```diff
@@ -26,42 +26,70 @@
   }
 }
 
+
+// Modal modifications to enable vertical scrolling with shadow overlays
+
+.modal-dialog {
+  @media(max-width: $screen-xs-max) {
+    height: calc(100% - 20px);
+  }
+}
+
 .modal-body {
   display: flex;
+  flex: 1 1 auto;
   flex-direction: column;
+  height: 100%;
   overflow-y: auto;
-  -webkit-overflow-scrolling: touch; // enable momentum scrolling in mobile Safari
+  padding: 0;
+  @include scroll-shadows-vertical;
+  -webkit-overflow-scrolling: touch;
 
-  @media(max-width: $screen-xs-max) {
-    bottom: 75px;
-    height: auto;
-    left: 11px;
-    position: fixed;
-    right: 11px;
-    top: 54px;
-    width: auto;
+  &.modal-body--centered {
+    @media(min-width: $screen-sm-min) {
```

Comment:
These changes seem undesirable for short modals?

![screen shot 2019-01-28 at 10 09 56 am](https://user-images.githubusercontent.com/895728/51845135-eae0ce00-22e4-11e9-8265-05a7b540f63a.png)

Reply:
<img width="684" alt="screen shot 2019-01-30 at 12 02 52 pm" src="https://user-images.githubusercontent.com/1874151/51998630-01268f80-2487-11e9-9935-f4db4be66e1b.png">